ServiceCapacityTracker/ServiceCapacityTrackerFunction/src/test/java/com/amazon/vas/servicecapacitytracker/component/ServiceCapacityDetailsComponentTest.java

```diff
@@ -0,0 +1,110 @@
+import com.amazon.vas.servicecapacitytracker.testdata.builders.DefaultServiceCapacityDetailsInputBOBuilder;
+import com.amazon.vas.servicecapacitytracker.testdata.builders.DefaultStoreCapacityDetailsBOBuilder;
+import com.amazon.vas.servicecapacitytracker.testdata.builders.DefaultStoreCapacityDetailsBOBuilderInputBuilder;
+import com.google.common.collect.ImmutableList;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class ServiceCapacityDetailsComponentTest {
+    @InjectMocks
+    private ServiceCapacityDetailsComponent serviceCapacityDetailsComponent;
+    @Mock
+    private MerchantDetailsBuilder merchantDetailsBuilder;
+    @Mock
+    private OfferDetailsBuilder offerDetailsBuilder;
+    @Mock
+    private StoreCapacityDetailsBOBuilder storeCapacityDetailsBOBuilder;
+    @Mock
+    private AppConfig appConfig;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testGetStoreCapacityDetails_whenValidInputIsPassedWithEmptyStoreName_thenSuccessfulResponse() {
+        final ServiceCapacityDetailsInputBO serviceCapacityDetailsInputBO =
+                new DefaultServiceCapacityDetailsInputBOBuilder().withEmptyStoreName().build();
+        final List<StoreCapacityDetailsBO> expectedStoreCapacityDetailsBOList =
+                ImmutableList.of(new DefaultStoreCapacityDetailsBOBuilder().forAggregatedMerchants().build());
+        List<StoreCapacityDetailsBOBuilderInput> storeCapacityDetailsBOBuilderInputList =
+                ImmutableList
+                        .of(new DefaultStoreCapacityDetailsBOBuilderInputBuilder().forAggregatedMerchants().build());
+        Mockito.when(appConfig.getCityMapper()).thenReturn(getDefaultCityMap());
+        Mockito.when(appConfig.getAsinMapper()).thenReturn(getDefaultAsinMap());
+        Mockito.when(storeCapacityDetailsBOBuilder.getResponse(
+                storeCapacityDetailsBOBuilderInputList, ConstantsClass.NUMBER_OF_COLUMNS))
+                .thenReturn(expectedStoreCapacityDetailsBOList);
+        List<StoreCapacityDetailsBO> storeCapacityDetailsBOList = serviceCapacityDetailsComponent
+                .getStoreCapacityDetails(serviceCapacityDetailsInputBO);
+        assertEquals(expectedStoreCapacityDetailsBOList, storeCapacityDetailsBOList);
+        Mockito.verify(storeCapacityDetailsBOBuilder)
+                .getResponse(storeCapacityDetailsBOBuilderInputList, ConstantsClass.NUMBER_OF_COLUMNS);
+    }
+
+    @Test
+    public void testGetStoreCapacityDetails_whenValidInputIsPassedWithStoreName_thenSuccessfulResponse() {
+        final ServiceCapacityDetailsInputBO serviceCapacityDetailsInputBO =
```

Comment:
segregate your test sections(i.e stubbing part, calling and verification/assertion part) by having a new line in between which increase readability. Applicable everywhere.